sqld/src/replication/primary/logger.rs

```diff
@@ -905,17 +913,22 @@ fn checkpoint_db(data_path: &Path) -> anyhow::Result<()> {
             &mut num_checkpointed as *mut _,
             std::ptr::null_mut(),
         );
-
-        // TODO: ensure correct page size
-        ensure!(
-            rc == 0 && num_checkpointed >= 0,
-            "failed to checkpoint database while recovering replication log"
-        );
-
-        conn.execute("VACUUM", ())?;
+        if rc == 0 {
+            if num_checkpointed == -1 {
+                Err(anyhow!(
+                    "Checkpoint failed: database journal_mode is not WAL"
+                ))
+            } else {
+                conn.execute("VACUUM", ())?;
```

Comment:
not sure a vacuum is necessary on every checkpoint

Reply:
That's the original code. I'm not sure, what would be the best course of action here. We can leave VACUUM as an explicit request from the user side.

Reply:
The intent was different, it was meant for replication log recovery

Reply:
Maybe I'll just remove it and let users perform it explicitly. It shouldn't be a problem most of the time.